plugins/event_source/webhook.py

```diff
@@ -4,49 +4,86 @@
 An ansible-rulebook event source module for receiving events via a webhook.
 
 Arguments:
-    host: The hostname to listen to. Set to 0.0.0.0 to listen on all
-          interfaces. Defaults to 127.0.0.1
-    port: The TCP port to listen to.  Defaults to 5000
+    host:     The hostname to listen to. Set to 0.0.0.0 to listen on all
+              interfaces. Defaults to 127.0.0.1
+    port:     The TCP port to listen to.  Defaults to 5000
+    token:    Authentication token expected from client
+    certfile: The optional path to a certificate file to enable TLS support
+    keyfile:  The optional path to a key file to be used together with certfile
+    password: The optional password to be used when loading the certificate chain
 
 """
 
 import asyncio
-from typing import Any, Dict
+import logging
+import ssl
+from typing import Any, Callable, Dict
 
 from aiohttp import web
 
+logger = logging.getLogger(__name__)
 routes = web.RouteTableDef()
 
 
 @routes.post(r"/{endpoint:.*}")
 async def webhook(request: web.Request):
     payload = await request.json()
     endpoint = request.match_info["endpoint"]
+    headers = dict(request.headers)
+    headers.pop("Authorization")
     data = {
         "payload": payload,
-        "meta": {"endpoint": endpoint, "headers": dict(request.headers)},
+        "meta": {"endpoint": endpoint, "headers": headers},
     }
     await request.app["queue"].put(data)
     return web.Response(text=endpoint)
 
 
+@web.middleware
+async def bearer_auth(request: web.Request, handler: Callable):
+    try:
+        scheme, token = request.headers["Authorization"].strip().split(" ")
+        if scheme != "Bearer":
+            raise web.HTTPUnauthorized(text="Only Bearer type is accepted")
+        elif token != request.app["token"]:
+            raise web.HTTPUnauthorized(text="Invalid authorization token")
+    except KeyError:
+        raise web.HTTPUnauthorized(reason="Missing authorization token")
+    except ValueError:
+        raise web.HTTPUnauthorized(text="Invalid authorization token")
+
+    return await handler(request)
+
+
 async def main(queue: asyncio.Queue, args: Dict[str, Any]):
-    app = web.Application()
+    app = web.Application(middlewares=[bearer_auth])
     app["queue"] = queue
+    app["token"] = args["token"]
```

Comment:
I really believe this param should be optional. First to keep the backward compatibility and also not all the clients are able to send the token.